mindmaps-engine/src/main/java/io/mindmaps/engine/controller/VisualiserController.java

```diff
@@ -105,11 +109,11 @@ private String matchQuery(Request req, Response res) {
             final JSONArray halArray = new JSONArray();
 
             withGraph(graph).parseMatch(req.queryParams(REST.Request.QUERY_FIELD))
-                    .stream()
+                    .parallelStream()
```

Comment:
Unfortunately the only way I know to confirm this is to monitor things with YourKit.

You could create a test that hits this line over and over again and while that tests runs check your memory usage. If it expands infinitely then we are likely opening transactions in the background.